### PR TITLE
[merp] Add an exception type for managed exceptions

### PR DESCRIPTION
**Pull request by alexischr**
Addresses https://github.com/mono/mono/issues/18681

### CODE REVIEW

File: mono/utils/mono-merp.c [hunk 1]
```diff
@@ -261,6 +262,8 @@ get_merp_exctype (MERPExcType exc)
 			return "0x04000000";
 		case MERP_EXC_HANG: 
 			return "0x02000000";
+		case MERP_EXC_MANAGED_EXCEPTION:
+			return "0x80000000";
```
Comment (lambdageek) [inline mono/utils/mono-merp.c]: @kdubau @alexischr Where do these magic hex numbers come from?

Comment (lambdageek) [reply mono/utils/mono-merp.c]: Kyle reminded me that this is just consumed as a string on their end. So this lgtm

**Comment by alexischr** [comment]
@monojenkins backport to 2020-02

**Comment by alexischr** [comment]
@monojenkins backport to 2019-12
@monojenkins backport to 2019-10

**Comment by alexischr** [comment]
Same unrelated failures as here: https://github.com/mono/mono/pull/18532#issuecomment-583228381

**Pull request merged**